### PR TITLE
ConfigMaps as alternatives to CRDs

### PR DESCRIPTION
**Pull request by FrancoAra**

### CODE REVIEW

File: baas-smoke-tests/src/test/resources/kubernetes/interactions-example-update-config-map.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: reserve-items
+  labels:
+    custom-resource-definition: interactions
+data:
+  image: baas-interaction-example-reserve-items:1.0.0
```
Comment (jacum) [inline baas-smoke-tests/src/test/resources/kubernetes/interactions-example-update-config-map.yaml:8]: how/where are `imagePullSecrets` specified?

File: baas-smoke-tests/src/test/resources/kubernetes/interactions-example-config-map.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+---
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: make-payment-and-ship-items
+  labels:
+    custom-resource-definition: interactions
+data:
+  image: interaction-make-payment-and-ship-items:1.0.0
```
[the remark this answers is not in the document]
Comment (FrancoAra) [reply baas-smoke-tests/src/test/resources/kubernetes/interactions-example-config-map.yaml:20]: That was my first approach, but my personal argument is, it is better to have a "first class" solution than a second class, that is we would be embedding one solution into another, which in my opinion also makes it cumbersome and "hackish" . Besides, dot property syntax is well known, I don't see too many issues with it.

File: baas-smoke-tests/src/test/resources/kubernetes/interactions-example-update-config-map.yaml (hunk 1 above)
Comment (FrancoAra) [reply baas-smoke-tests/src/test/resources/kubernetes/interactions-example-update-config-map.yaml:8]: Secrets can be either mounted or added to environment variables, and taken from the Kubernetes secrets store. The `interactions-example-config-map.yaml` has some secrets configured as example

File: baas-smoke-tests/src/test/resources/kubernetes/interactions-example-config-map.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (FrancoAra) [reply baas-smoke-tests/src/test/resources/kubernetes/interactions-example-config-map.yaml:20]: But if we start getting issues with it we could add embedded CRDs as a 3rth option I would propose

**Pull request merged**